hikari/impl/entity_factory.py

```diff
@@ -2542,7 +2542,7 @@ def deserialize_command_interaction(
             guild_id=guild_id,
             guild_locale=locales.Locale(payload["guild_locale"]) if "guild_locale" in payload else None,
             locale=locales.Locale(payload["locale"]),
-            channel_id=snowflakes.Snowflake(payload["channel_id"]),
+            channel=self.deserialize_partial_channel(payload["channel"]),
```

Comment:
the channel objects here include a lot more fields than just the partial channel's fields

e.g.
```json
{
    "type": 0,
    "topic": null,
    "theme_color": null,
    "rate_limit_per_user": 0,
    "position": 20,
    "permissions": "140737488355327",
    "parent_id": "561885448721399808",
    "nsfw": false,
    "name": "dwads",
    "last_message_id": "1114922806820479057",
    "id": "717586538925523004",
    "icon_emoji": {
        "name": "\ud83e\udd16",
        "id": null
    },
    "guild_id": "561884984214814744",
    "flags": 0,
    "default_auto_archive_duration": 60
}
```
```json
{
    "type": 1,
    "last_message_id": "1096465893028003910",
    "id": "463185043796262922",
    "flags": 0
}
```

Reply:
This was also mentioned in the PR which documented this field https://github.com/discord/discord-api-docs/pull/6051#discussion_r1156415144, just another case of Discord having shit docs tbh